core/cas-server-core-authentication-mfa-api/src/test/java/org/apereo/cas/authentication/RestMultifactorAuthenticationProviderBypassEvaluatorTests.java

```diff
@@ -67,4 +67,31 @@ public void verifyOperationFailsWithNoProvider() {
             assertTrue(res);
         }
     }
+
+    @Test
+    public void verifyRestSendsQueryParameters() {
+        val applicationContext = new StaticApplicationContext();
+        applicationContext.refresh();
+        ApplicationContextProvider.holdApplicationContext(applicationContext);
+        ApplicationContextProvider.registerBeanIntoApplicationContext(applicationContext,
+            MultifactorAuthenticationPrincipalResolver.identical(), UUID.randomUUID().toString());
+
+        try (val webServer = new MockWebServer(9316,
```

Comment:
Could you explain how this test actually "verifies rest sends query parameters"?

It looks like it does not. You'll need to design a test that actually tests the functionality you put together. This test does more or less nothing to that effect.

Reply:
Agreed, it should ideally verify from the mock server's perspective that the query parameters are received. 

What would be the best way to go about this? I can think of 3 ways:
  - Add functionality to the existing `MockWebServer` class so that it collects request(s) received and exposes an interface to retrieve the received requests and their attributes
  - Subclass `MockWebServer` to provide the above functionality
  - Use `MockWebServer` from `okhttp3` - they already seem to provide the functionality

Any input is appreciated!

(I'm rather inexperienced with java and spring, please excuse my naive proposals)

Reply:
I implemented option 3 discussed above (using `okhttp3.mockwebserver` to record params on the webserver side). However, that involves adding the `okhttp3:mockwebserver` dependency under `tests`

If you have a chance to look at the newest commit, let me know what you think of this approach